homeassistant/components/bouncie/sensor.py

```diff
@@ -0,0 +1,203 @@
+
+    entities = [
+        BouncieSensor(coordinator=coordinator, description=description)
+        for description in sensor_types
+    ]
+
+    async_add_entities(entities, False)
+
+
+class BouncieSensor(CoordinatorEntity[BouncieDataUpdateCoordinator], SensorEntity):
+    """Bouncie sensor."""
+
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator: BouncieDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Init the BouncieSensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._state = None
+        self._attrs: dict[str, str] = {}
+        self._attr_name = f"{description.name}"
+        self._attr_unique_id = f"{coordinator._client_id}-{description.key.lower()}"
+        self._vehicle_info = next(
+            (
+                v
+                for v in self.coordinator.data["vehicles"]
+                if v["vin"] == self.entity_description.key.split("-")[-1]
+            ),
+            None,
+        )
+
+    def _update_car_info_state(self):
+        self._state = (
+            "Running" if self._vehicle_info["stats"]["isRunning"] else "Not Running"
+        )
+
+    def _update_car_info_attributes(self):
+        self._attrs[const.ATTR_VEHICLE_MODEL_MAKE_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["make"]
+        self._attrs[const.ATTR_VEHICLE_MODEL_NAME_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["name"]
+        self._attrs[const.ATTR_VEHICLE_MODEL_YEAR_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["year"]
+        self._attrs[const.ATTR_VEHICLE_NICKNAME_KEY] = self._vehicle_info["nickName"]
+        self._attrs[const.ATTR_VEHICLE_STANDARD_ENGINE_KEY] = self._vehicle_info[
+            "standardEngine"
+        ]
+        self._attrs[const.ATTR_VEHICLE_VIN_KEY] = self._vehicle_info["vin"]
+        self._attrs[const.ATTR_VEHICLE_IMEI_KEY] = self._vehicle_info["imei"]
+        self._update_car_stats_attributes()
+
+    def _update_car_odometer_state(self):
+        self._state = int(self._vehicle_info["stats"]["odometer"])
+
+    def _update_car_stats_attributes(self):
+        self._attrs[const.ATTR_VEHICLE_STATS_LAST_UPDATED_KEY] = self._vehicle_info[
+            "stats"
+        ]["lastUpdated"]
+
+    def _update_car_address_state(self):
+        self._state = self._vehicle_info["stats"]["location"]["address"]
+
+    def _update_car_fuel_level_state(self):
+        self._state = int(self._vehicle_info["stats"]["fuelLevel"])
+
+    def _update_car_speed_state(self):
+        self._state = self._vehicle_info["stats"]["speed"]
+
+    def _update_car_mil_state(self):
+        self._state = self._vehicle_info["stats"]["mil"]["milOn"]
+
+    def _update_car_mil_attributes(self):
+        self._state = self._vehicle_info["stats"]["mil"]["lastUpdated"]
+
+    def _update_car_battery_state(self):
```

Comment:
What is the purpose of breaking all of these state updates into individual methods?

Reply:
In future releases, these methods will have some calculation on the values.